### PR TITLE
Determine latest component versions automatically

### LINKED ISSUE #12: Dynamically determine version of knative components
[Source](https://github.com/knative-sandbox/kn-plugin-quickstart/pull/11#discussion_r652308952)

Currently, we have hard-coded the Knative versions into the source code (see [here](https://github.com/knative-sandbox/kn-plugin-quickstart/blob/main/pkg/install/install.go#L22-L23)). 

Ideally we should dynamically query the version of the latest release. Or, potentially this is something that's handled by the update bot (?).

### PR DESCRIPTION
**Pull request by psschwei**
Signed-off-by: Paul S. Schweigert <paul@paulschweigert.com>

# Changes

- Update Knative component versions automatically at build time, similar to the logic used by the client in https://github.com/knative/client/pull/1625

Fixes #12 

/assign @dsimansk @rhuss

### CODE REVIEW

File: hack/build-flags.sh [hunk 1]
```diff
@@ -24,6 +24,11 @@ function build_flags() {
     [[ -n "${commit}" ]] || (echo "error getting the current commit" && exit 1)
     version="v$(date +%Y%m%d)-local-${commit}"
   fi
+  # Knative component versions
+  local serving="`git ls-remote --tags --ref https://github.com/knative/serving.git | cut -d '-' -f2 | cut -d 'v' -f2 | sort -Vr | head -n 1`"
+  local kourier="`git ls-remote --tags --ref https://github.com/knative-sandbox/net-kourier.git | cut -d '-' -f2 | cut -d 'v' -f2 | sort -Vr | head -n 1`"
+  local eventing="`git ls-remote --tags --ref https://github.com/knative/eventing.git | cut -d '-' -f2 | cut -d 'v' -f2 | sort -Vr | head -n 1`"
```
Comment (dprotaso) [inline hack/build-flags.sh:30]: I think this would break with release branches - doesn't this always pick the latest?

Comment (psschwei) [reply hack/build-flags.sh:30]: Good catch

Comment (psschwei) [reply hack/build-flags.sh]: tested fix on 1.6 and 1.7 branches, works as expected

**Pull request merged**